### DISCUSSION TITLE
Textual supporting I18n

### DISCUSSION THREAD

**Discussion by AAraKKe** [Textualize/textual#3263] [Ideas]
## The problem statement
Textual goal is to be able to generate beautiful tuis with as wide support as possible and following Rich's principle of ease of use.

For this reason it seems kind of off-topic to think about Textual supporting localization internally. However, few interface frameworks do not support this or do not have a way of enabling it. 

Python provides options to handle translations ([gettext](https://docs.python.org/3/library/gettext.html#module-gettext) for instance) and I am sure there are other third party options. However, it seems to me that it would be awesome if Textual would add support for internationalization with the same principles and through Textual objects in a way that hides away all the implementations detail. The user would provide translations files with the text that needs to be translated and the App would expose a binding to switch languages seamlessly when requested.

## Usecase
I am coming to this proposal because I have an use case for this and while for now I am just working in Spanish I think this would be a nice tool also for non-spanish speakers. I have been thinking about adding support for multiple languages and it always seems that, if Textual had a way of allowing this (as Rails has for instance), it would be so much nicer than any implementation I can build into it.

What do you guys think?

**Comment by willmcgugan** [comment]
I think we certainly should provide localization and internationalization support. It probably isn't a great deal of work for the core library, but it would mean that folk wouldn't need to roll their own.

**Comment by AAraKKe** [comment]
Sounds great! Is there any ongoing effort for that or a design discussion? If I can help in any way I can dedicate some of my time to it (cannot commit to a lot because of work but I can certainly help).
